forge-game/src/main/java/forge/game/trigger/TriggerDamageDoneOnce.java

```diff
@@ -36,6 +37,21 @@ public boolean performTest(Map<AbilityKey, Object> runParams) {
             return false;
         }
 
+        if (hasParam("SingleTarget")) {
+            Object target = runParams.get(AbilityKey.DamageTarget);
+            final Map<Card, Integer> damageMap = (Map<Card, Integer>) runParams.get(AbilityKey.DamageMap);
+            for (Card source : getDamageSources(damageMap)) {
```

Comment:
if you use DamageDone trigger instead you wouldn't need to iterate over all sources?